### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by andriy-shymkiv**
[DEMO LINK](https://andriy-shymkiv.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,47 +15,106 @@ export const goodsFromServer = [
...
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={sortType === SortType.ALPABET
+            ? 'button is-info'
+            : 'button is-info is-light'}
```
Comment (sh1gatsu) [inline src/App.tsx:74]: Better use classNames i guess, but its subjective ;)

**Review by sh1gatsu — APPROVED** [review]
GJ 🔥

File: src/App.tsx [hunk 2]
```diff
@@ -15,47 +15,106 @@ export const goodsFromServer = [
...
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.NONE:
+        return 0;
```
Comment (denys-danyliuk) [inline src/App.tsx:44]:
```suggestion
        break;
```

File: src/App.tsx [hunk 3]
```diff
@@ -15,47 +15,106 @@ export const goodsFromServer = [
...
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setValue] = useState(false);
```
Comment (denys-danyliuk) [inline src/App.tsx:59]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,47 +15,106 @@ export const goodsFromServer = [
...
+  const [sortType, setCount] = useState(SortType.NONE);
```
Comment (denys-danyliuk) [inline src/App.tsx:60]:
```suggestion
  const [sortType, setSortType] = useState(SortType.NONE);
```
you should name functions like setNameOfTheProperty

File: src/App.tsx (hunk 1 above)
Comment (denys-danyliuk) [reply src/App.tsx:74]: agree, use classNames

File: src/App.tsx [hunk 5]
```diff
@@ -15,47 +15,106 @@ export const goodsFromServer = [
...
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || sortType !== SortType.NONE) && (
```
Comment (denys-danyliuk) [inline src/App.tsx:100]: you can move condition to some variable

File: src/App.tsx [hunk 6]
```diff
@@ -15,47 +15,106 @@ export const goodsFromServer = [
...
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {orderedGoods.map(good => (
+            <li data-cy="Good" key={good}>{good}</li>
```
Comment (denys-danyliuk) [inline src/App.tsx:116]:
```suggestion
            <li data-cy="Good" key={good}>
              {good}
            </li>
```

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (Andriy-Shymkiv) [reply src/App.tsx]: i can't just break it because i got some TS errors

Comment (denys-danyliuk) [reply src/App.tsx]: got you, then return 0

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**